crypto/kzg/src/lib.rs

```diff
@@ -174,20 +192,30 @@ impl Kzg {
 
     /// Computes the cells and associated proofs for a given `blob` at index `index`.
     pub fn compute_cells_and_proofs(&self, blob: &Blob) -> Result<CellsAndKzgProofs, Error> {
-        let blob_bytes: &[u8; BYTES_PER_BLOB] = blob
-            .as_ref()
-            .try_into()
-            .expect("Expected blob to have size {BYTES_PER_BLOB}");
+        if self.use_ckzg {
+            let (cells, proofs) =
+                c_kzg::Cell::compute_cells_and_kzg_proofs(blob, &self.trusted_setup)?;
 
-        let (cells, proofs) = self
-            .context
-            .prover_ctx()
-            .compute_cells_and_kzg_proofs(blob_bytes)
-            .map_err(Error::ProverKZG)?;
+            let cells = cells.map(|c| Box::new(c.to_bytes()));
+            let proofs = proofs.map(|p| KzgProof::from(p.to_bytes().into_inner()));
 
-        // Convert the proof type to a c-kzg proof type
-        let c_kzg_proof = proofs.map(KzgProof);
-        Ok((cells, c_kzg_proof))
+            Ok((cells, proofs))
+        } else {
+            let blob_bytes: &[u8; BYTES_PER_BLOB] = blob
+                .as_ref()
+                .try_into()
+                .expect("Expected blob to have size {BYTES_PER_BLOB}");
+
+            let (cells, proofs) = self
+                .context
+                .prover_ctx()
+                .compute_cells_and_kzg_proofs(blob_bytes)
+                .map_err(Error::ProverKZG)?;
+
+            let c_kzg_proof = proofs.map(KzgProof);
+
+            Ok((cells, c_kzg_proof))
```

Comment:
This should be plural. Plus minor `c_kzg` vs `ckzg` nit. Going with `ckzg`, like `use_ckzg`.

```suggestion
            let ckzg_proofs = proofs.map(KzgProof);

            Ok((cells, ckzg_proofs))
```